cdap-ui/app/cdap/components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/index.tsx

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Dialog, DialogContent, DialogTitle, IconButton } from '@material-ui/core';
+import withStyles, { StyleRules, WithStyles } from '@material-ui/core/styles/withStyles';
+import CloseIcon from '@material-ui/icons/Close';
+import If from 'components/If';
+import { h2Styles } from 'components/Markdown/MarkdownHeading';
+import { WIDGET_TYPE_TO_ATTRIBUTES } from 'components/PluginJSONCreator/constants';
+import WidgetAttributeInput from 'components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetAttributesCollection/WidgetAttributeInput';
+import WidgetInput from 'components/PluginJSONCreator/Create/Content/WidgetCollection/WidgetInput';
+import { ICreateContext } from 'components/PluginJSONCreator/CreateContextConnect';
+import * as React from 'react';
+
+const styles = (theme): StyleRules => {
+  return {
+    widgetAttributeInput: {
+      '& > *': {
+        width: '100%',
+        marginTop: '10px',
+        marginBottom: '10px',
+      },
+    },
+    widgetAttributesTitle: {
+      marginTop: '15px',
+      marginBottom: '15px',
+    },
+    h2Title: {
+      ...h2Styles(theme).root,
+      marginBottom: '5px',
+    },
+  };
+};
+
+interface IWidgetAttributesCollectionProps extends WithStyles<typeof styles>, ICreateContext {
+  widgetAttributesOpen: boolean;
+  onWidgetAttributesClose: () => void;
+  widgetID: number;
+}
+
+const WidgetAttributesCollectionView: React.FC<IWidgetAttributesCollectionProps> = ({
+  classes,
+  widgetAttributesOpen,
+  onWidgetAttributesClose,
+  widgetID,
+  widgetToInfo,
+  setWidgetToInfo,
+  widgetToAttributes,
+  setWidgetToAttributes,
+}) => {
+  const widget = widgetToInfo[widgetID];
+  const widgetType = widget ? widget.widgetType : null;
+  const attributeFields =
+    widgetToAttributes && widgetToAttributes[widgetID]
+      ? Object.keys(widgetToAttributes[widgetID])
+      : [];
+  return (
+    <div>
+      <Dialog
```

Comment:
We should use the standard modal we have within cdap. I don't believe we have this modal ported to css-in-js yet but it should be easy to add that. IT is a simple modal with dark blue backeground and white text for heading. Like this,

<img width="1026" alt="Screen Shot 2020-06-02 at 10 19 06 PM" src="https://user-images.githubusercontent.com/1452845/83598639-19953700-a51f-11ea-8895-ffa0349b2abf.png">

Reply:
Sounds good. I'll be adding a separate PR to add this modal, since this will involve changing code outside of the current project.